src/data.cpp

```diff
@@ -86,7 +86,7 @@ int Data::isEmpty()
       ret = 1;
       break;
     case Data::TypeString:
-      if( val == NULL || val->length() <= 0 ) {
+      if( val == NULL || val->length() <= 0 || *val == "null" ) {
```

Comment:
I think this may be better handled in the specification parsing code, as I don't think it's optimal to generally treat a string containing "null" as empty. Unfortunately, libyaml, being a c, not a c++, library, is fairly low level and doesn't handle any type conversions automatically. See e.g. #18.

This (probably) isn't an issue with the PHP extension as it has it's own conversion code using native PHP types.

I've opened another PR in #26 but let me know if you disagree or if it breaks your use-case or w/e.